File: Beginner/1021_Banknotes_and_Coins/.swift
import Foundation

if let input = readLine(), let n = Double(input) {
    var a = n * 100
    let b = Int(a / 10000)
    a = a.truncatingRemainder(dividingBy: 10000)
    let c = Int(a / 5000)
    a = a.truncatingRemainder(dividingBy: 5000)
    let d = Int(a / 2000)
    a = a.truncatingRemainder(dividingBy: 2000)
    let e = Int(a / 1000)
    a = a.truncatingRemainder(dividingBy: 1000)
    let f = Int(a / 500)
    a = a.truncatingRemainder(dividingBy: 500)
    let g = Int(a / 200)
    a = a.truncatingRemainder(dividingBy: 200)
    let h = Int(a / 100)
    a = a.truncatingRemainder(dividingBy: 100)
    let i = Int(a / 50)
    a = a.truncatingRemainder(dividingBy: 50)
    let j = Int(a / 25)
    a = a.truncatingRemainder(dividingBy: 25)
    let k = Int(a / 10)
    a = a.truncatingRemainder(dividingBy: 10)
    let l = Int(a / 5)
    a = a.truncatingRemainder(dividingBy: 5)

    print("NOTAS:")
    print("\(b) nota(s) de R$ 100.00")
    print("\(c) nota(s) de R$ 50.00")
    print("\(d) nota(s) de R$ 20.00")
    print("\(e) nota(s) de R$ 10.00")
    print("\(f) nota(s) de R$ 5.00")
    print("\(g) nota(s) de R$ 2.00")
    print("MOEDAS:")
    print("\(h) moeda(s) de R$ 1.00")
    print("\(i) moeda(s) de R$ 0.50")
    print("\(j) moeda(s) de R$ 0.25")
    print("\(k) moeda(s) de R$ 0.10")
    print("\(l) moeda(s) de R$ 0.05")
    print("\(Int(a)) moeda(s) de R$ 0.01")
}

File: Beginner/1021_Banknotes_and_Coins/.ts
import * as fs from 'fs';

const input: string = fs.readFileSync('/dev/stdin', 'utf8');
const [a] = input.split(' ');
let valor: number = parseFloat(a);
const cedula: number[] = [100.00, 50.00, 20.00, 10.00, 5.00, 2.00, 1.00, 0.50, 0.25, 0.10, 0.05, 0.01];
let aux: number = 0;

console.log("NOTAS:");
for (let i = 0; i < 6; i++) {
    aux = Math.floor(valor / cedula[i]);
    console.log(`${aux} nota(s) de R$ ${cedula[i].toFixed(2)}`);
    valor = valor % cedula[i];
}

console.log("MOEDAS:");
for (let j = 6; j < cedula.length; j++) {
    aux = Math.floor(valor / cedula[j]);
    console.log(`${aux} moeda(s) de R$ ${cedula[j].toFixed(2)}`);
    valor = (valor % cedula[j]) + 0.00001;
}
